engo_sdl.go

```diff
@@ -157,8 +130,9 @@ func RunIteration() {
 		for event := sdl.PollEvent(); event != nil; event = sdl.PollEvent() {
 			switch e := event.(type) {
 			case *sdl.QuitEvent:
-				Exit()
-				return
+				go func() {
+					closeGame <- struct{}{}
```

Comment:
Why didn't we just use `close(closeGame)`?

Reply:
Exit calls close on it already, if it hits that and it's already closed then you'll get a panic: close of closed channel

Reply:
Writing to a closed channel will also panic though. Unless I am missing something.

Reply:
True. I'll try using sync.Once so it only closes one time in case it's hit multiple times, then I can call exit here safely.

Reply:
I think `sync.Once` is a great plan. :)